src/libraries/System.Net.Sockets/tests/FunctionalTests/CreateSocketTests.cs

```diff
@@ -445,6 +446,25 @@ public async Task Ctor_SafeHandle_Listening_Success(bool shareSafeHandle)
             }
         }
 
+        [DllImport("libc")]
+        static extern int socket(int domain, int type, int protocol);
```

Comment:
do you have specific example @stephentoub? I looked at #34861 and I did not see any failures besides comment from @lpereira.  I did test with socketpair and I did not see any issues aside from weird EndPoints.
I can add test  as well.

Reply:
https://github.com/dotnet/runtime/pull/34861#discussion_r408170184
At least based on that, we're not getting a peer name and thus not setting _isConnected to true, which means that change isn't working on macOS.  Is that now fixed with your changes?

Reply:
yes, I saw that comment but I did not observe any failures - either in the PR or in my local tests. I did tests with socketpair and everything was working for me. But I don't know if that is enough. I would need to see the failure to be sure. I can make build with #34861 changes but I don't know how to trigger it - I would expect that running tests would be sufficient?  But that does pass in CI.

Reply:
You wrote in another comment "This is because _rightEndPoint is alwasys set on copy."  That wasn't previously the case with socketpair on macOS, so maybe something in your PR fixed it.

Reply:
I added basic test for socketpair. We can extend that if we see failures.